src/main/sensors/gyro_init.c

```diff
@@ -612,94 +605,103 @@ bool gyroInit(void)
         break;
     }
 
-    gyroDetectionFlags = GYRO_NONE_MASK;
+    gyroDetectedFlags = 0;
     uint8_t gyrosToScan = gyroConfig()->gyrosDetected;
+    // scan all gyros if gyrosDetected is zero
+    if (gyrosToScan == 0) {
+        gyrosToScan = 0xff;
+    }
+    // always scan gyro_enabled_bitmask
+    gyrosToScan |= gyroConfig()->gyro_enabled_bitmask;
 
-    gyro.gyroToUse = gyroConfig()->gyro_to_use;
     gyro.gyroDebugAxis = gyroConfig()->gyro_filter_debug_axis;
 
-    if ((!gyrosToScan || (gyrosToScan & GYRO_1_MASK)) && gyroDetectSensor(&gyro.gyroSensor1, gyroDeviceConfig(0))) {
-        gyroDetectionFlags |= GYRO_1_MASK;
-    }
-
-#if defined(USE_MULTI_GYRO)
-    if ((!gyrosToScan || (gyrosToScan & GYRO_2_MASK)) && gyroDetectSensor(&gyro.gyroSensor2, gyroDeviceConfig(1))) {
-        gyroDetectionFlags |= GYRO_2_MASK;
+    for (int i = 0; i < GYRO_COUNT; i++) {
+        // Only attempt to detect a gyro if it's enabled or we're doing an auto-scan
+        if (gyrosToScan & GYRO_MASK(i)) {
+            if (gyroDetectSensor(&gyro.gyroSensor[i], gyroDeviceConfig(i))) {
+                // If we detected a gyro, make sure it's in the enabled bitmask
+                // This ensures that during first boot, all detected gyros are enabled
+                gyroDetectedFlags |= GYRO_MASK(i);
+            }
+        }
     }
-#endif
 
-    if (gyroDetectionFlags == GYRO_NONE_MASK) {
+    if (gyroDetectedFlags == 0) {
         return false;
     }
 
     bool eepromWriteRequired = false;
     if (!gyrosToScan) {
```

Comment:
Fix for `cliStatus`:
```suggestion
    if (gyrosToScan) {
```